plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergColumnHandle.java

```diff
@@ -34,17 +36,27 @@
 {
     private final ColumnIdentity columnIdentity;
     private final Type type;
+    private final ColumnIdentity baseColumnIdentity;
+    private final Type baseType;
     private final Optional<String> comment;
+    private final List<String> path;
+    private List<String> fullPath;
 
     @JsonCreator
     public IcebergColumnHandle(
             @JsonProperty("columnIdentity") ColumnIdentity columnIdentity,
             @JsonProperty("type") Type type,
-            @JsonProperty("comment") Optional<String> comment)
+            @JsonProperty("baseColumnIdentity") ColumnIdentity baseColumnIdentity,
+            @JsonProperty("baseType") Type baseType,
+            @JsonProperty("comment") Optional<String> comment,
+            @JsonProperty("path") List<String> path)
```

Comment:
I feel we may be able to get away with storing less data here, especially duplicated columnIdentity and baseColumnIdentity. 

Ultimately, we may want to just start sending the serialized schema in IcebergTableHandle, and use `Schema` methods for field access. That'd save us from a lot of "reinventing the wheel" w.r.t. nested field access. (Not suggesting that this PR needs to be blocked on that.) cc @electrum 

For now, is it possible to simplify this if we use a map instead of a list in column identity? Where the ordering is required, we can use Trino type's fields. 

Also, I think it's better to store iceberg ids as much as possible (e.g. instead of strings in `path`), for robustness.

Reply:
Made a few changes: 
- Updated IcebergColumnHandle to just store the base column identity
- Updated the path to use int iceberg field ids instead of Strings
- Included a change to the `ColumnIdentity#children` representation to allow for quick lookup by index or by field id